codegen/src/shuttle_main/mod.rs

```diff
@@ -85,15 +85,11 @@ impl Parse for BuilderOption {
 
 impl Loader {
     pub(crate) fn from_item_fn(item_fn: &mut ItemFn) -> Option<Self> {
-        let fn_ident = item_fn.sig.ident.clone();
-
-        if fn_ident.to_string().as_str() == "main" {
-            emit_error!(
-                fn_ident,
-                "shuttle_runtime::main functions cannot be named `main`"
-            );
-            return None;
-        }
+        // rename function to allow any name, such as 'main'
```

Comment:
Nit: we use capitalized comments elsewhere, so it would be good to be consistent.